lib/jobs/ipmi-job.js

```diff
@@ -223,6 +223,71 @@ function ipmiJobFactory(
         });
     };
 
+
+    /**
+     * Recursive function to traverse the sel log
+     * @memberOf IpmiJob
+     *
+     * @param selArray
+     * @param selRecordID
+     * @param data
+     * @param counter
+     * @param totalEntries
+     */
+    IpmiJob.prototype.loop = function(selArray, selRecordID, data, counter, totalEntries) {
+        var self = this;
+        var dataCloned = (JSON.parse(JSON.stringify(data)));//cloning the data object
+        var parsedEntry;
+        
+        return self.getSelEntries(data, selRecordID)
+        .then(function(unparsedSelEntry) {
+            parsedEntry = parser.parseSelDataEntries(unparsedSelEntry);
+            var recordID = parsedEntry[0]["SEL Record ID"];
+            var recordType = parsedEntry[0]["Record Type"];
+            if (recordType === "02") {
+                return self.getRawSelEntry(dataCloned, recordID)
+                    .then(function(rawRead){
+                        // we add two properties ("Sensor Type Code" and "Event Type Code") to each sel entry.
+                        var decodedSel = {
+                            "Event Type Code" : "",
+                            "Sensor Type Code" : ""
+                        };
+                        var regex = new RegExp('\r?\n','g');
+                        var rawRead = rawRead.replace(regex, '');
+                        var values = rawRead.split(" ");
+                        var mask = 1 << 7; // a mask to clear the 7th bit
+                        values = values.slice(1,19);// remove empty entry
+                        var rawSelRecord = values.slice(2,18);
+                        var nextRecordID = values.slice(0,2);//This is the next SEL entry
+                        //This is from table 32 of the IPMI spec v4
+                        var pad = "00";
+                        var eventType = (parseInt(rawSelRecord[12],16) & ~mask).toString(16).slice(-2);
```

Comment:
@tannoa2 you could add '/*jslint bitwise: true */' to silence the warnings '&' ad '~'